---
title: "iUSEiSEE - About this workshop"
author: "Federico Marini^[marinif@uni-mainz.de], Najla Abassi^[abassina@uni-mainz.de] <br><hr>
  Collection of material developed by the iSEE core dev team <br>
  (Kevin Rue-Albrecht, Federico Marini, Charlotte Soneson, Aaron Lun)"
output: 
    html_document:
      theme: cosmo
      toc: yes
      toc_float: yes
      number_sections: true
vignette: >
  %\VignetteIndexEntry{00. About this workshop}
  %\VignettePackage{iUSEiSEE}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
bibliography: iUSEiSEErefs.bib
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
suppressPackageStartupMessages({
  library("BiocStyle")
})
```


<img src="iSEE.png" align="right" alt="" width="120" />

# Description

The iSEE package [@Rue-Albrecht2018-isee] is a flexible, powerful and extendible application to explore any dataset stored in SummarizedExperiment objects, including single cell, spatially resolved data, but also bulk data (transcriptomics and proteomics being just two examples of what can efficiently fit in there!).

iSEE enables a multitude of panels to gain in-depth insight in your data; it does so combining interactivity and reproducibility, and providing an ideal companion to other tools (such as CELLxGENE) for deeper dives into data.

This package provides a complete overview of the functionality of the `r Biocpkg("iSEE")` Bioconductor package, and is intended as some kind of "masterclass-level" workshop - indeed, the material is enough to cover a full-day workshop on iSEE.

# Pre-requisites

* Familiarity with bulk or single-cell sequencing data (ideally scRNA-seq)
* Familiarity with the SummarizedExperiment/SingleCellExperiment class, or 
alternatively the Seurat or AnnData frameworks - basically, any of the widely used 
ecosystem for working with/storing/processing single cell data
* Familiarity with single-cell data visualizations and representations

# Participation

Participants will be able (and heartily invited) to follow along in the live coding parts of the workshop. This is very much intended as a hands-on learning resource.

Given the hands-on nature of this, this is going to be a BYOD (Bring Your Own Device) event - if cloud-based setups are provided, this information will be passed on in advance to the participants.
Nonetheless, we value the side effect of participants installing a working environment as a part of the learning objectives.

Participants will (likely) use a set of stickers to indicate their progress throughout the material, to be attached onto the laptop lid. In brief:

* Yellow sticker: "I am working on the task"
* Green sticker: "I am done with the task"
* Red sticker: "I am stuck on this, could you please come help me?"

We do stress the importance of working together on such tasks, therefore participants are very encouraged to talk to their neighbors during the task sessions!

# _R_ / _Bioconductor_ packages used

* `iSEE`: https://bioconductor.org/packages/iSEE
* `iSEEu`: https://bioconductor.org/packages/iSEEu
* `iSEEde`: https://bioconductor.org/packages/iSEEde
* `iSEEpathways`: https://bioconductor.org/packages/iSEEpathways
* `iSEEhub`: https://bioconductor.org/packages/iSEEhub
* `iSEEindex`: https://bioconductor.org/packages/iSEEindex
* `iSEEfier`: https://bioconductor.org/packages/iSEEfier


# Outline of the course

After a short presentation, we will start going through the material in the `vignettes` folder:

* [iUSEiSEE](iUSEiSEE_01.Rmd) - a tl;dr version of the whole, which can be used as a compact summary of the content
* [Overview on iSEE](iSEE_overview_02.Rmd) - a detailed overview on iSEE and its functionality
* [The iSEE cookbook](iSEE_quickrecipes_03.Rmd) - a set of self-contained recipes with simple-to-advanced tasks one can complete within iSEE
* [Bonus advanced content around iSEE](bonus_content_04.Rmd) - some additional content, focused on some advanced or more specific tasks to carry out with iSEE

# Workshop goals and objectives

## Learning goals

* Outline the features of iSEE, built upon the RStudio Shiny framework
* Utilize interactive GUI components and layouts to efficiently extract information from biological data sets
* Describe how to construct interactive apps and custom panels
* Demonstrate how these package can smoothly be integrated into and enhance existing workflows for the datasets at hand

## Learning objectives

* Set up a local environment for running iSEE apps
* Interact with components of the iSEE user interface to visually inspect and discuss various data sets
* Identify and locate configurable aspects of iSEE apps
* Practice interactive visualization over a single-cell RNA-sequencing workflow
* Imagine use cases and future developments for interactive visualization as part of computational workflows
* Design and create iSEE instances for collections of datasets

# References {-}
