---
title: "About this workshop"
author: "Federico Marini^[marinif@uni-mainz.de], Kevin Rue-Albrecht^[kevinrue67@gmail.com], Charlotte soneson^[charlottesoneson@gmail.com], Aaron Lun^[infinite.monkeys.with.keyboards@gmail.com], Najla Abassi^[abassina@uni-mainz.de]"
output: 
    html_document:
      theme: cosmo
      toc: yes
      toc_float: yes
      number_sections: true
vignette: >
  %\VignetteIndexEntry{00. About this workshop}
  %\VignettePackage{iUSEiSEE}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
bibliography: iUSEiSEErefs.bib
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
suppressPackageStartupMessages({
  library("BiocStyle")
})
```


<img src="iSEE.png" align="right" alt="" width="120" />

# Description

The `r Biocpkg("iSEE")` Bioconductor package [@Rue-Albrecht2018-isee] is a flexible, powerful and extensible application to explore any dataset stored in a `r Biocpkg("SummarizedExperiment")` object, including single cell and spatially resolved data, but also bulk data (such as transcriptomics and proteomics) and many types of non-biological data.

`r Biocpkg("iSEE")` enables a multitude of panels to gain in-depth insight in your data; it does so combining interactivity and reproducibility, and providing an ideal companion to other tools for deeper dives into data.

This package provides an overview of the functionality of the `r Biocpkg("iSEE")` package. 
The material is equally suitable for use in a "masterclass-style" workshop as for self-study.

# Pre-requisites

Most of the examples in this workshop will center around bulk and single-cell RNA sequencing data. 
For this reason, a certain background knowledge about these types of data and the questions that are typically addressed with them can be helpful to get the most out of the content. 
More precisely, the following pre-requisities are useful:

* Familiarity with bulk or single-cell sequencing data (ideally scRNA-seq)
* Familiarity with the `r Biocpkg("SummarizedExperiment")`/`r Biocpkg("SingleCellExperiment")` class, or alternatively the `r CRANpkg("Seurat")` or AnnData frameworks - basically, any of the widely used ecosystem for working with/storing/processing single cell data
* Familiarity with single-cell data visualizations and representations

# Participation

Participants will be encouraged to follow along in the live coding parts of the workshop - this is very much intended as a hands-on learning resource.

# _R_ / _Bioconductor_ packages used

* `iSEE`: https://bioconductor.org/packages/iSEE
* `iSEEu`: https://bioconductor.org/packages/iSEEu
* `iSEEde`: https://bioconductor.org/packages/iSEEde
* `iSEEpathways`: https://bioconductor.org/packages/iSEEpathways
* `iSEEhub`: https://bioconductor.org/packages/iSEEhub
* `iSEEindex`: https://bioconductor.org/packages/iSEEindex
* `iSEEfier`: https://bioconductor.org/packages/iSEEfier

# Outline of the workshop

The following material is available in the `vignettes` folder:

* [Preparation](d01_preparation.Rmd) - instructions for installing the required packages and downloading the example data
* [Overview of iSEE](d02_iSEE_overview.Rmd) - a detailed overview of `r Biocpkg("iSEE")` and its functionality
* [The iSEE cookbook](d03_iSEE_quickrecipes.Rmd) - a set of self-contained recipes with simple-to-advanced tasks one can complete within `r Biocpkg("iSEE")`
* [Bonus advanced content around iSEE](d04_bonus_content.Rmd) - some additional content, focused on some advanced or more specific tasks to carry out with `r Biocpkg("iSEE")`
* [Extending iSEE](d05_extending_iSEE.Rmd) - material focused on extending `r Biocpkg("iSEE")` by creating new panel classes

# Workshop goals and objectives

## Learning goals

* Outline the features of `r Biocpkg("iSEE")`, built upon the R/Shiny framework
* Utilize interactive GUI components and layouts to efficiently extract information from biological data sets
* Describe how to construct interactive apps and custom panels
* Demonstrate how `r Biocpkg("iSEE")` and related packages can smoothly be integrated into and enhance existing workflows for the datasets at hand

## Learning objectives

* Set up a local environment for running `r Biocpkg("iSEE")` apps
* Interact with components of the `r Biocpkg("iSEE")` user interface to visually inspect and discuss various datasets
* Identify and locate configurable aspects of `r Biocpkg("iSEE")` apps
* Practice interactive visualization over a single-cell RNA-sequencing workflow
* Imagine use cases and future developments for interactive visualization as part of computational workflows
* Design and create `r Biocpkg("iSEE")` instances for collections of datasets

# Additional resources {#additional-resources}

A lot of materials have been developed, from the first use of `r Biocpkg("iSEE")` to contributions of new panel classes in separate downstream packages.

* The iSEE website: https://isee.github.io/
* Publication (F1000Research, 2018): https://f1000research.com/articles/7-741/v1
* Bioconductor landing page: https://bioconductor.org/packages/iSEE/
* Development version (bug reports etc): https://github.com/iSEE/iSEE
* Material from previous workshops: https://isee.github.io/resources.html#workshops
* Deployed examples: https://isee.github.io/apps.html, https://github.com/iSEE/iSEE_instances
* Custom panel examples: https://github.com/iSEE/iSEE_custom
* The bookdown book on _Extending iSEE_: https://isee.github.io/iSEE-book/

# References {-}
