---
title: "iUSEiSEE - An overview of iSEE"
author: "Federico Marini^[marinif@uni-mainz.de], Najla Abassi^[abassina@uni-mainz.de] <br><hr>
  Collection of material developed by the iSEE core dev team <br>
  (Kevin Rue-Albrecht, Federico Marini, Charlotte Soneson, Aaron Lun)"
output: 
    html_document:
      theme: cosmo
      toc: yes
      toc_float: yes
      number_sections: true
vignette: >
  %\VignetteIndexEntry{02. Overview of iSEE}
  %\VignettePackage{iUSEiSEE}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
bibliography: iUSEiSEErefs.bib
---

<img src="iSEE.png" align="right" alt="" width="120" />

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  cache = FALSE,
  collapse = TRUE,
  comment = "#>",
  crop = NA
)
suppressPackageStartupMessages({
  library("BiocStyle")
  library("iSEE")
})
htmltools::tagList(rmarkdown::html_dependency_font_awesome())
```

```{r, eval=!exists("SCREENSHOT"), include=FALSE}
SCREENSHOT <- function(x, ...) knitr::include_graphics(x)
```


# Introduction

This vignette provides an overview of the graphical interface of `r Biocpkg("iSEE")` applications [@Rue-Albrecht2018-isee].
To follow along, make sure that you launch the default `r Biocpkg("iSEE")` instance as described in the next code block at the start of the [following section](#default-app).

Note that in the default configuration, the panels do not look exactly like the ones shown in the screenshots that you will see below.
For example, data points are not immediately colored, and the default annotation variables displayed by each panel may differ.
The [hands-on recipes](recipes.html) workshop of this workshop will demonstrate how to modify the content of the panels, and how they are displayed. 

Note that for simplicity, we typically refer to a `SummarizedExperiment` in this workshop; however, `r Biocpkg("iSEE")` works seamlessly for objects of any class extending `SummarizedExperiment` as well (e.g., `SingleCellExperiment`, `DESeqDataSet`).
That said, some types of panels -- such as the `r .fullName(ReducedDimensionPlot())` -- are only available for objects that contain a `reducedDim` slot (in particular, `SingleCellExperiment` objects); the basic `SummarizedExperiment` class does not contain this slot.
In this workshop, we refer to the rows of the `SummarizedExperiment` object as 'features' (these can be genes, transcripts, genomic regions, etc) and to the columns as 'samples' (which, in our example data set, are single cells).

# The default iSEE app {#default-app}

Using the demonstration data set, we can launch an `r Biocpkg("iSEE")` instance for exploring this data set using the `iSEE()` function without any further argument.
This will produce an app using the default configuration; that is, the app instance will include one panel of each built-in class for which the relevant information is available in the `SummarizedExperiment` object.

```{r, message=FALSE}
library("iSEE")
sce_location <- system.file("datasets", "sce_pbmc3k.RDS", package = "iUSEiSEE")
sce_location
sce <- readRDS(sce_location)
app <- iSEE(sce)
```

```{r runApp, eval=FALSE}
shiny::runApp(app)
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-initial.png", delay = 40)
```

The main argument to the `iSEE()` function is a `SummarizedExperiment` object, or an object of any class extending `SummarizedExperiment` (such as `SingleCellExperiment`, in this case).
No other restrictions are made on the type of data stored in the object, and `r Biocpkg("iSEE")` can be used for interactive visualization of many different types of data.
It is also worth noting that for various types of data, Bioconductor packages provides functionality for directly importing quantifications generated by external software packages into a `SummarizedExperiment` object.
For example, the `r Biocpkg("DropletUtils")` package can read quantifications from the 10x Genomics CellRanger pipeline for single-cell RNA-seq data, and the `r Biocpkg("tximeta")` package can be used to read data from transcript quantification pipelines into a `SummarizedExperiment` object.

While we will not make explicit use of it in this workshop, we note that it is also possible to call `iSEE()` without providing a `SummarizedExperiment` object. 
In that case, the user will be prompted to upload such an object, serialized into a `.rds` file. 
It is also possible to import a specification of the initial panel setup. 

```{r}
app <- iSEE()
```

```{r, eval=FALSE}
shiny::runApp(app)
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-no-se.png", delay = 10)
```

# The built-in panel types {#user-interface}

## The panel hierarchy

The `r Biocpkg("iSEE")` user interface (UI) consists of a number of panels, each displaying the data provided in the `SummarizedExperiment` from a specific perspective.
There are 8 standard panel types; 6 plot panels and 2 table panels, all showcased in the figure shown at the start of the [previous section](#default-app).
In the default configuration, one panel of each type is included when launching the `r Biocpkg("iSEE")` user interface.
However, users are free to rearrange, resize, remove or add panels freely, as demonstrated in the separate vignette of [workshop recipes](recipes.html).
We provide a brief overview of each panel type in the following subsections.

In addition to the 8 standard panel types included in `r Biocpkg("iSEE")`, users can create custom panels (both plots and tables).
Moreover, the `r Biocpkg("iSEEu")` (iSEE universe) package contains additional panel types. 
The creation and configuration of custom panels is also demonstrated in a series of [workshop advanced recipes](recipes.html).

Specifically, `r Biocpkg("iSEE")` panels are implemented as a hierarchy of S4 classes.
Only _concrete_ classes (orange) may be instantiated as panels in the user interface; while _virtual_ classes (grey) provide functionality shared by families of panels (e.g., sample-oriented plots, feature-oriented tables).

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/panel-classes.png", dpi = 180)
```

<!--
This is perhaps not very pedagogical - the app will not look like the one below if it is just opened with the object created above.
But it's nice to show the capabilities of the panels too, not just the default settings.
I added a few sentences to the introduction of this section to try to address this.
--> 

## `r .fullName(ReducedDimensionPlot())` {#reddimplot}

The reduced dimension plot can display any reduced dimension representation that is present in the `reducedDim` slot of the `SingleCellExperiment` object.

Note that this slot is not defined for the base `SummarizedExperiment` class, in which case the user interface does not allow the inclusion of panels of this type.
 
```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ReducedDimensionPlot(
  PanelWidth = 12L,
  Type = "TSNE", ColorBy = "Column data",
  ColorByColumnData = "Cluster"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-reduceddimensionplot.png", delay = 10)
```

## `r .fullName(ColumnDataPlot())` {#coldataplot}

The column data plot can display one or two of the provided column annotations (from the `colData` slot).
Depending on the class of the selected annotations, the panel shows either a scatter plot, a violin plot, or a Hinton diagram [@Hinton1991-hintondiagram; @Bremner1994-hintonplots].

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ColumnDataPlot(
  PanelWidth = 12L,
  XAxis = "Column data", XAxisColumnData = "Cluster",
  YAxis = "log10_total"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-columndataplot.png", delay = 10)
```

## `r .fullName(RowDataPlot())` {#rowdataplot}

Analogous to the [column data plot](#coldataplot) above, the row data plot displays one or two of the provided row annotations (from the `rowData` slot).
Depending on the class of the selected annotations, the panel displays either a scatter plot, a violin plot, or a Hinton plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(RowDataPlot(
  PanelWidth = 12L, XAxis = "Row data",
  YAxis = "detected", XAxisRowData = "log10_total"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-rowdataplot.png", delay = 10)
```

## `r .fullName(ComplexHeatmapPlot())` {#heatmap}

The complex heatmap panel displays, for any assay, the observed values for a subset of the features across the samples.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ComplexHeatmapPlot(
  PanelWidth = 12L, Assay = "logcounts", CustomRows = TRUE,
  CustomRowsText = c("CD14", "CD27", "CD52", "CD68", "CD19", "CD4"),
  ColumnData = "Cluster", AssayCenterRows = TRUE
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-complexheatmapplot.png", delay = 10)
```

## `r .fullName(FeatureAssayPlot())` {#featassayplot}

The feature assay plot displays the observed values for one feature across the samples.
It is also possible to plot the observed values for two features, in a scatter plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(FeatureAssayPlot(
  PanelWidth = 12L, XAxis = "Column data",
  XAxisColumnData = "Cluster", YAxisFeatureName = "CD27",
  Assay = "logcounts", ColorBy = "Column data",
  ColorByColumnData = "Cluster"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-featureassayplot.png", delay = 10)
```

## `r .fullName(SampleAssayPlot())` {#sampassayplot}

Analogous to the [`r .fullName(FeatureAssayPlot())`](#featassayplot) above, the `r .fullName(SampleAssayPlot())` shows the observed values for all features, for one of the samples.
It is also possible to plot the observed values for two samples, in a scatter plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(SampleAssayPlot(
  PanelWidth = 12L, XAxis = "Sample name",
  XAxisSampleName = "Cell1", YAxisSampleName = "Cell2",
  PointSize = 2
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-sampleassayplot.png", delay = 10)
```

## `r .fullName(RowDataTable())` {#rowdatatable}

The row data table displays all information provided in the `rowData` slot of the `SummarizedExperiment` object, leveraging the interactivity provided by the `r CRANpkg("DT")` package.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(RowDataTable(PanelWidth = 12L)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-rowdatatable.png", delay = 10)
```

## `r .fullName(ColumnDataTable())` {#coldatatable}

Analogous to the [`r .fullName(RowDataTable())`](#rowdatatable) above, the column data table displays all information provided in the `colData` slot of the `SummarizedExperiment` object.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ColumnDataTable(PanelWidth = 12L)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-columndatatable.png", delay = 10)
```

# Collapsible boxes with display controls {#collapsible-boxes}

## Data parameters {#data-parameters-box}

Each plot panel type has a `Data parameters` collapsible box.
This box has different content for each panel type, but in all cases it lets the user control the data that is displayed in the plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ColumnDataPlot(
  PanelWidth = 12L, DataBoxOpen = TRUE,
  XAxis = "Column data", XAxisColumnData = "Cluster",
  YAxis = "log10_total"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-columndataplot-dataparameters.png", delay = 10)
```

## Visual parameters {#visual-parameters-box}

In contrast to the `Data parameters` collapsible box that lets users control _what_ is displayed in the plot, the `Visual parameters` box lets users control _how_ the information is displayed.

This collapsible box contains the controls to change the size, shape, opacity, and color of the points, to facet the plot by any available categorical annotation, to subsample points for increased speed of plot rendering, and to control how legends are displayed.

```{r, message=FALSE, echo=FALSE}
app <- iSEE(sce, initial = list(ColumnDataPlot(
  PanelWidth = 12L, VisualBoxOpen = TRUE,
  XAxis = "Column data", XAxisColumnData = "Cluster",
  YAxis = "log10_total",
  VisualChoices = c("Color", "Point", "Facet", "Shape", "Size", "Text")
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-columndataplot-visualparameters.png", delay = 10)
```

## Selection parameters {#selection-parameter-box}

The `Selection parameters` collapsible box provides controls to transfer selections of points (features or samples) between panels.

We demonstrate examples of point transmission in the separate vignette of [workshop recipes](recipes.html)

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ReducedDimensionPlot(
  PanelWidth = 12L, SelectionBoxOpen = TRUE,
  Type = "TSNE", ColorBy = "Column data",
  ColorByColumnData = "Cluster"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("images/isee-reduceddimensionplot-selectionparameters.png", delay = 10)
```

# Additional controls {#additional-controls}

At the top-right corner of the iSEE application, users can find additional controls for reproducibility, configuration, and help.

```{r, echo = FALSE, eval = TRUE}
panel_organization_fig <- "images/PanelOrganization.png"
export_fig <- "images/Export.png"
documentation_fig <- "images/Documentation.png"
additional_info_fig <- "images/AdditionalInfo.png"
```

|              |            |                                        |
|--------------|:----------:|----------------------------------------|
| Organization | `r knitr::include_graphics(panel_organization_fig, dpi = 180)` | [Organize panels](#organize-panels) <br> [Examine panel chart](#examine-panel-chart) |
| Export        | `r knitr::include_graphics(export_fig, dpi = 180)`       | [Download panel output](#download-output) <br> [Extract R code](#extract-r-code) <br> [Display panel settings](#display-panel-settings) |
| Documentation      | `r knitr::include_graphics(documentation_fig, dpi = 180)`     | [Quick tour](#quick-tour) <br> [Open the vignette](#browse-vignettes) |
| Additional info    | `r knitr::include_graphics(additional_info_fig, dpi = 180)`    | [About this session](#app-session-info) <br> [About iSEE](#about-isee) |

## Organize panels {#organize-panels}

As mentioned above, the default behaviour of the `iSEE()` function is to launch an instance of the user interface that displays one panel of each of the standard types (provided the underlying data is available, e.g. for reduced dimension plots the `reducedDim` slot is required).
However, in some cases it is desirable to have multiple panels of the same type, and/or exclude some panel types.

In order to accommodate such situations, users can add, remove, change the order of and resize all panels via the `Organization` menu in the top-right corner.

Clicking in the selectize box listing all current panels will present you with a drop-down menu from which you can choose additional panels to add.
Similarly, panels can be removed by clicking on the <i class="fa fa-times"></i> icon associated with the panel name.

Each panel can be individually resized by changing the width and height.
Note that the total width of a row in the interface is 12 units.
When the width of a panel is greater than the space available, the panel is moved to a new row.

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/panel-organization.png", dpi = 180)
```

## Linking panels and transmitting point selections {#examine-panel-chart}

When exploring data, it is often useful to be able to select a subset of points and investigate different aspects of their characteristic features.
In `r Biocpkg("iSEE")`, this can be achieved by _selecting_ points in one panel, and _transmitting_ this selection to one or more other panels.

The brushing and point selection can also be programmatically preconfigured.

The button `Examine panel chart` display a graph that reports any active point transmission between panels in the app.

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/panel-graph.png", dpi = 180)
```

## Download panel output {#download-output}

The `Download panel output` button opens a modal window listing all the current panels in the app.
Checkboxes allow users to select any subset of panels to export.
Finally, clicking the `Download` button in that modal will prompt the app to save plots to PDF files, tables to CSV files, package the set of files in a ZIP archive that users can download and save on their computer.

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/panel-download.png", dpi = 180)
```

## Extract R code {#extract-r-code}

The fact that data exploration is done interactively is no reason to forego reproducibility! 
To this end, `r Biocpkg("iSEE")` lets you export the exact R code used to create each of the currently visible plots.

Importantly, the script reported by `r Biocpkg("iSEE")` contains a short preamble needed to set up variable names that are used in individual panels, including active brushes used to transfer point selections.

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/code-tracker.png", dpi = 180)
```

## Display panel settings {#display-panel-settings}

It can take a great amount of time to achieve a satisfactory panel configuration.
To avoid the need to manually organize the panels each time the app is opened, `r Biocpkg("iSEE")` offers the possibility to export code that can be reused later to
programmatically specify how the app is initialized, as well as to inspect and export the current panel settings for future use.

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/panel-settings.png", dpi = 180)
```

## Quick tour {#quick-tour}

One important aspect of visualization is the ability to share your insights with others. 
A powerful way of easily getting people unfamiliar with your data up to speed is to provide a walkthrough of the interface and the different types of plots that are displayed. 
With `r Biocpkg("iSEE")`, this can be achieved using _tours_. 
To configure a tour, you need to create a text file with two columns; named `element` and `intro`, with the former describing the UI element to highlight in each step, and the latter providing the descriptive text that will be displayed.

Clicking the `Quick tour` button launches an interactive tour of the interface using the `r BiocStyle::CRANpkg("rintrojs")` package, highlighting specific elements of the user interface, labeled with information and instructions guiding users through panels and tasks specific to individual apps.

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/quick-tour.png", dpi = 180)
```

You can see a live example of a more complex tour in action, visit <https://marionilab.cruk.cam.ac.uk/iSEE_pbmc4k/>.
The code used to create this app and the associated tour is available from <https://github.com/iSEE/iSEE2018/tree/master/tours/pbmc4k>.

## Open the vignette {#browse-vignettes}

The user interface navigation bar also includes this button to open the introductory vignette to `r Biocpkg("iSEE")` in your we browser.

Depending on the version of `r Biocpkg("iSEE")` that you are using, this will adaptively lead you to a locally built vignette present on your computer, or the _release_ or the _devel_ version of the Bioconductor package landing page, e.g. <`r BiocStyle::Biocpkg("iSEE", vignette = "basic.html", label = paste("Bioconductor", BiocManager::version()))`>.

## Session information {#app-session-info}

This button displays the output of `sessionInfo()`, which is a useful piece of information to report when reporting an issue with an app.

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("images/session-info.png", dpi = 180)
```

## About iSEE {#about-isee}

This button provides information about the authors of `r Biocpkg("iSEE")`, as well as citation information.

If you use this package, please use the following citation information:

> Rue-Albrecht K, Marini F, Soneson C, Lun ATL (2018). “iSEE: Interactive SummarizedExperiment Explorer.” _F1000Research_, *7*,
> 741. doi: 10.12688/f1000research.14966.1 (URL: https://doi.org/10.12688/f1000research.14966.1).

A BibTeX entry for LaTeX users is:

```
@Article{,
  title = {iSEE: Interactive SummarizedExperiment Explorer},
  author = {Kevin Rue-Albrecht and Federico Marini and Charlotte Soneson and Aaron T. L. Lun},
  publisher = {F1000 Research, Ltd.},
  journal = {F1000Research},
  year = {2018},
  month = {Jun},
  volume = {7},
  pages = {741},
  doi = {10.12688/f1000research.14966.1},
}
```


# Modes and additional panels

The `r Biocpkg("iSEEu")` ("iSEE universe") Bioconductor package defines additional custom panels and predefined 'modes' (startup configurations) that may be useful for specific applications. 
Here we illustrate the use of the _reduced dimension_ mode, which will start an application with one reduced dimension panel for each reduced dimension representation in the input object.

```{r}
library("iSEEu")
app <- modeReducedDim(sce)
```

```{r, eval = FALSE}
shiny::runApp(app)
```

```{r, echo=FALSE, out.width = '100%'}
SCREENSHOT("images/isee-iseeu-modereduceddim.png", delay=10)
```

Furthermore, since `r Biocpkg("iSEE")` version 2.0.0, users can leverage the implementation of panels as a hierarchy of S4 classes to rapidly extend the framework and develop new types of panels with virtually unlimited freedom of functionality.
In this framework, new panel types can be readily integrated in apps alongside built-in panel types and immediately benefit of most panel functionality with minimal effort, including the capacity to transmit selections to and from other panels.
New panels may be implemented by extending the core virtual class `Panel` directly for full control over the user interface and reactive observers;
alternatively, it is often desirable for developers to inherit from one of the concrete panel classes to get most of the essential functionality from the parent class "for free".

Use cases demonstrating the implementation of new panels with various levels of complexity are available in the [Extending iSEE](https://isee.github.io/iSEE-book/) bookdown.
For example, the _Hexagonal reduced dimension plot_ - implemented in the `r Biocpkg("iSEEu")` package - demonstrates an alternative to the downsampling strategy, by summarizing data points into hexagonal bins.

```{r}
app <- iSEE(sce, initial = list(
  ReducedDimensionPlot(PanelWidth = 6L),
  ReducedDimensionHexPlot(PanelWidth = 6L)
))
```

```{r, eval = FALSE}
shiny::runApp(app)
```
```{r, echo=FALSE, out.width = '100%'}
SCREENSHOT("images/isee-iseeu-modereduceddimhex.png", delay=10)
```


Another useful mode you can use with your data, especially if using iSEE for mass cytometry data, would be the `modeGating`. 
This launches an app preconfigured with multiple chain-linked feature expression plots for interactive data exploration.

```{r}
# Select top variable genes ----
plot_count <- 6
rv <- rowVars(assay(sce, "logcounts"))
top_var <- head(order(rv, decreasing=TRUE), plot_count*2)
top_var_genes <- rownames(sce)[top_var]

plot_features <- data.frame(
  x=head(top_var_genes, plot_count),
  y=tail(top_var_genes, plot_count),
  stringsAsFactors=FALSE
)

# launch the app itself ----

app <- modeGating(sce, 
                  plotAssay = "logcounts",
                  features = plot_features)
```

```{r, eval = FALSE}
shiny::runApp(app)
```

```{r, echo=FALSE, out.width = '100%'}
SCREENSHOT("images/isee-iseeu-modegating.png", delay=20)
```

Interestingly, iSEEu also has a set of extra panels that can come in handy in different situations.
For example, `iSEEu::AggregatedDotPlot` implements an aggregated dot plot where each feature/group combination is represented by a dot. 
The color of the dot scales with the mean assay value across all samples for a given group, while the size of the dot scales with the proportion of non-zero values across samples in that group.
This can be an alternative to a heatmap, and an example can be seen in the chunks below:

```{r}
app <- iSEE(
  sce, 
  initial = list(
    AggregatedDotPlot(
      Assay = "logcounts",
      ColumnDataLabel = "labels_main",
      CustomRowsText = top_var_genes
    )
    
  )
)
```

```{r, echo=FALSE, out.width = '100%'}
SCREENSHOT("images/isee-iseeu-aggrodotplot.png", delay=20)
```


New panels and modes may be defined in independent _R_ packages that declare `Imports: iSEE` in their DESCRIPTION file, and distributed online to create an "iSEE-verse" of interoperable packages for interactive visualization, akin to the `r CRANpkg('tidyverse')` collection of R packages for data science.
If you have ideas about additional panels and/or modes that might be useful, we welcome issues and pull requests at https://github.com/iSEE/iSEEu.




# Additional resources {#additional-resources}

A lot of materials have been developed, from the first use of `r Biocpkg("iSEE")` to contributions of new panel classes in separate downstream packages.

* The bookdown book on _Extending iSEE_: https://isee.github.io/iSEE-book/
* Bioconductor landing page: https://bioconductor.org/packages/iSEE/
* Publication (F1000Research, 2018): https://f1000research.com/articles/7-741/v1
* Deployed examples: https://marionilab.cruk.cam.ac.uk/, code at https://github.com/iSEE/iSEE2018
  * Further deployments in the https://github.com/iSEE/iSEE_instances repo
  * More and more examples: https://isee.github.io/apps.html
* Custom panel examples: https://github.com/iSEE/iSEE_custom
* Development version (bug reports etc): https://github.com/iSEE/iSEE

# Session info {-}

```{r}
sessionInfo()
```

# References {-}
