<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>An overview of iSEE • iUSEiSEE</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="An overview of iSEE">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">iUSEiSEE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-workshop-sessions" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Workshop sessions</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-workshop-sessions">
<li><a class="dropdown-item" href="../articles/d00_about_this_workshop.html">00. About this workshop</a></li>
    <li><a class="dropdown-item" href="../articles/d01_preparation.html">01. Preparation</a></li>
    <li><a class="dropdown-item" href="../articles/d02_iSEE_overview.html">02. An overview of iSEE</a></li>
    <li><a class="dropdown-item" href="../articles/d03_iSEE_quickrecipes.html">03. The iSEE cookbook</a></li>
    <li><a class="dropdown-item" href="../articles/d04_bonus_content.html">04. Bonus iSEE content</a></li>
    <li><a class="dropdown-item" href="../articles/d05_extending_iSEE.html">05. Extending iSEE</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/iSEE/iUSEiSEE"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="d02_iSEE_overview_files/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="d02_iSEE_overview_files/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>An overview of iSEE</h1>
                        <h4 data-toc-skip class="author">Federico
Marini<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:marinif@uni-mainz.de" class="email"&gt;marinif@uni-mainz.de&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>,
Kevin Rue-Albrecht<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:kevinrue67@gmail.com" class="email"&gt;kevinrue67@gmail.com&lt;/a&gt;&lt;/p&gt;'><sup>2</sup></a>, Charlotte soneson<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:charlottesoneson@gmail.com" class="email"&gt;charlottesoneson@gmail.com&lt;/a&gt;&lt;/p&gt;'><sup>3</sup></a>, Aaron Lun<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:infinite.monkeys.with.keyboards@gmail.com" class="email"&gt;infinite.monkeys.with.keyboards@gmail.com&lt;/a&gt;&lt;/p&gt;'><sup>4</sup></a>, Najla
Abassi<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:abassina@uni-mainz.de" class="email"&gt;abassina@uni-mainz.de&lt;/a&gt;&lt;/p&gt;'><sup>5</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iUSEiSEE/blob/main/vignettes/d02_iSEE_overview.Rmd" class="external-link"><code>vignettes/d02_iSEE_overview.Rmd</code></a></small>
      <div class="d-none name"><code>d02_iSEE_overview.Rmd</code></div>
    </div>

    
    
<p><img src="iSEE.png" align="right" alt="" width="120"></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette provides an overview of the graphical interface of
<em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>
applications <span class="citation">(Rue-Albrecht et al. 2018)</span>.
To follow along, make sure that you launch the default <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>
instance as described in the next code block at the start of the <a href="#default-app">following section</a>.</p>
<p>Note that in the default configuration, the panels do not look
exactly like the ones shown in the screenshots that you will see below.
For example, data points are not immediately colored, and the default
annotation variables displayed by each panel may differ. The <a href="03_iSEE_quickrecipes.html">hands-on recipes</a> workshop of this
workshop will demonstrate how to modify the content of the panels, and
how they are displayed.</p>
<p>Note that for simplicity, we typically refer to a
<code>SummarizedExperiment</code> in this workshop; however, <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> works
seamlessly for objects of any class extending
<code>SummarizedExperiment</code> as well (e.g.,
<code>SingleCellExperiment</code>, <code>DESeqDataSet</code>). That
said, some types of panels – such as the Reduced dimension plot – are
only available for objects that contain a <code>reducedDim</code> slot
(in particular, <code>SingleCellExperiment</code> objects); the basic
<code>SummarizedExperiment</code> class does not contain this slot. In
this workshop, we refer to the rows of the
<code>SummarizedExperiment</code> object as ‘features’ (these can be
genes, transcripts, genomic regions, etc) and to the columns as
‘samples’ (which, in our example data set, are single cells).</p>
</div>
<div class="section level2">
<h2 id="default-app">The default iSEE app<a class="anchor" aria-label="anchor" href="#default-app"></a>
</h2>
<p>Using the demonstration data set, we can launch an <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>
instance for exploring this data set using the <code><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE()</a></code>
function without any further argument. This will produce an app using
the default configuration; that is, the app instance will include one
panel of each built-in class for which the relevant information is
available in the <code>SummarizedExperiment</code> object.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://isee.github.io/iSEE/" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="va">sce_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"datasets"</span>, <span class="st">"sce_pbmc3k.RDS"</span>, package <span class="op">=</span> <span class="st">"iUSEiSEE"</span><span class="op">)</span></span>
<span><span class="va">sce_location</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmpwKgIu6/temp_libpath67033657d0a3/iUSEiSEE/datasets/sce_pbmc3k.RDS"</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">sce_location</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/02-isee-initial.png" width="100%"></p>
<p>The main argument to the <code><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE()</a></code> function is a
<code>SummarizedExperiment</code> object, or an object of any class
extending <code>SummarizedExperiment</code> (such as
<code>SingleCellExperiment</code>, in this case). No other restrictions
are made on the type of data stored in the object, and <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> can be
used for interactive visualization of many different types of data. It
is also worth noting that for various types of data, Bioconductor
packages provides functionality for directly importing quantifications
generated by external software packages into a
<code>SummarizedExperiment</code> object. For example, the <em><a href="https://bioconductor.org/packages/3.21/DropletUtils" class="external-link">DropletUtils</a></em>
package can read quantifications from the 10x Genomics CellRanger
pipeline for single-cell RNA-seq data, and the <em><a href="https://bioconductor.org/packages/3.21/tximeta" class="external-link">tximeta</a></em>
package can be used to read data from transcript quantification
pipelines into a <code>SummarizedExperiment</code> object.</p>
<p>While we will not make explicit use of it in this workshop, we note
that it is also possible to call <code><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE()</a></code> without providing a
<code>SummarizedExperiment</code> object. In that case, the user will be
prompted to upload such an object, serialized into a <code>.rds</code>
file. It is also possible to import a specification of the initial panel
setup.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/02-isee-no-se.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="user-interface">The built-in panel types<a class="anchor" aria-label="anchor" href="#user-interface"></a>
</h2>
<div class="section level3">
<h3 id="the-panel-hierarchy">The panel hierarchy<a class="anchor" aria-label="anchor" href="#the-panel-hierarchy"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> user
interface (UI) consists of a number of panels, each displaying the data
provided in the <code>SummarizedExperiment</code> from a specific
perspective. There are 8 standard panel types; 6 plot panels and 2 table
panels, all showcased in the figure shown at the start of the <a href="#default-app">previous section</a>. In the default configuration,
one panel of each type is included when launching the <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> user
interface. However, users are free to rearrange, resize, remove or add
panels freely, as demonstrated in the separate vignette of <a href="03_iSEE_quickrecipes.html">workshop recipes</a>. We provide a
brief overview of each panel type in the following subsections.</p>
<p>In addition to the 8 standard panel types included in <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>, users
can create custom panels (both plots and tables). Moreover, the <em><a href="https://bioconductor.org/packages/3.21/iSEEu" class="external-link">iSEEu</a></em> (iSEE
universe) package contains additional panel types. The creation and
configuration of custom panels is also demonstrated in a series of <a href="03_iSEE_quickrecipes.html">workshop advanced recipes</a>.</p>
<p>Specifically, <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> panels
are implemented as a hierarchy of S4 classes. Only <em>concrete</em>
classes (orange) may be instantiated as panels in the user interface;
while <em>virtual</em> classes (grey) provide functionality shared by
families of panels (e.g., sample-oriented plots, feature-oriented
tables).</p>
<p><img src="images/panel-classes.png" width="100%"></p>
<!--
This is perhaps not very pedagogical - the app will not look like the one below if it is just opened with the object created above.
But it's nice to show the capabilities of the panels too, not just the default settings.
I added a few sentences to the introduction of this section to try to address this.
-->
</div>
<div class="section level3">
<h3 id="reddimplot">Reduced dimension plot<a class="anchor" aria-label="anchor" href="#reddimplot"></a>
</h3>
<p>The reduced dimension plot can display any reduced dimension
representation that is present in the <code>reducedDim</code> slot of
the <code>SingleCellExperiment</code> object.</p>
<p>Note that this slot is not defined for the base
<code>SummarizedExperiment</code> class, in which case the user
interface does not allow the inclusion of panels of this type.</p>
<p><img src="images/02-isee-reduceddimensionplot.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="coldataplot">Column data plot<a class="anchor" aria-label="anchor" href="#coldataplot"></a>
</h3>
<p>The column data plot can display one or two of the provided column
annotations (from the <code>colData</code> slot). Depending on the class
of the selected annotations, the panel shows either a scatter plot, a
violin plot, or a Hinton diagram <span class="citation">(Hinton and
Shallice 1991; Bremner, Gotts, and Denham 1994)</span>.</p>
<p><img src="images/02-isee-columndataplot.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="rowdataplot">Row data plot<a class="anchor" aria-label="anchor" href="#rowdataplot"></a>
</h3>
<p>Analogous to the <a href="#coldataplot">column data plot</a> above,
the row data plot displays one or two of the provided row annotations
(from the <code>rowData</code> slot). Depending on the class of the
selected annotations, the panel displays either a scatter plot, a violin
plot, or a Hinton plot.</p>
<p><img src="images/02-isee-rowdataplot.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="heatmap">Complex heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>The complex heatmap panel displays, for any assay, the observed
values for a subset of the features across the samples.</p>
<p><img src="images/02-isee-complexheatmapplot.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="featassayplot">Feature assay plot<a class="anchor" aria-label="anchor" href="#featassayplot"></a>
</h3>
<p>The feature assay plot displays the observed values for one feature
across the samples. It is also possible to plot the observed values for
two features, in a scatter plot.</p>
<p><img src="images/02-isee-featureassayplot.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="sampassayplot">Sample assay plot<a class="anchor" aria-label="anchor" href="#sampassayplot"></a>
</h3>
<p>Analogous to the <a href="#featassayplot">Feature assay plot</a>
above, the Sample assay plot shows the observed values for all features,
for one of the samples. It is also possible to plot the observed values
for two samples, in a scatter plot.</p>
<p><img src="images/02-isee-sampleassayplot.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="rowdatatable">Row data table<a class="anchor" aria-label="anchor" href="#rowdatatable"></a>
</h3>
<p>The row data table displays all information provided in the
<code>rowData</code> slot of the <code>SummarizedExperiment</code>
object, leveraging the interactivity provided by the <em><a href="https://CRAN.R-project.org/package=DT" class="external-link">DT</a></em> package.</p>
<p><img src="images/02-isee-rowdatatable.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="coldatatable">Column data table<a class="anchor" aria-label="anchor" href="#coldatatable"></a>
</h3>
<p>Analogous to the <a href="#rowdatatable">Row data table</a> above,
the column data table displays all information provided in the
<code>colData</code> slot of the <code>SummarizedExperiment</code>
object.</p>
<p><img src="images/02-isee-columndatatable.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="collapsible-boxes">Collapsible boxes with display controls<a class="anchor" aria-label="anchor" href="#collapsible-boxes"></a>
</h2>
<div class="section level3">
<h3 id="data-parameters-box">Data parameters<a class="anchor" aria-label="anchor" href="#data-parameters-box"></a>
</h3>
<p>Each plot panel type has a <code>Data parameters</code> collapsible
box. This box has different content for each panel type, but in all
cases it lets the user control the data that is displayed in the
plot.</p>
<p><img src="images/02-isee-columndataplot-dataparameters.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="visual-parameters-box">Visual parameters<a class="anchor" aria-label="anchor" href="#visual-parameters-box"></a>
</h3>
<p>In contrast to the <code>Data parameters</code> collapsible box that
lets users control <em>what</em> is displayed in the plot, the
<code>Visual parameters</code> box lets users control <em>how</em> the
information is displayed.</p>
<p>This collapsible box contains the controls to change the size, shape,
opacity, and color of the points, to facet the plot by any available
categorical annotation, to subsample points for increased speed of plot
rendering, and to control how legends are displayed.</p>
<p><img src="images/02-isee-columndataplot-visualparameters.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="selection-parameter-box">Selection parameters<a class="anchor" aria-label="anchor" href="#selection-parameter-box"></a>
</h3>
<p>The <code>Selection parameters</code> collapsible box provides
controls to transfer selections of points (features or samples) between
panels.</p>
<p>We demonstrate examples of point transmission in the separate
vignette of <a href="03_iSEE_quickrecipes.html">workshop recipes</a></p>
<p><img src="images/02-isee-reduceddimensionplot-selectionparameters.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="additional-controls">Additional controls<a class="anchor" aria-label="anchor" href="#additional-controls"></a>
</h2>
<p>At the top-right corner of the iSEE application, users can find
additional controls for reproducibility, configuration, and help.</p>
<table class="table">
<colgroup>
<col width="21%">
<col width="18%">
<col width="60%">
</colgroup>
<tbody>
<tr class="odd">
<td>Organization</td>
<td align="center"><img src="images/PanelOrganization.png" width="433"></td>
<td>
<a href="#organize-panels">Organize panels</a> <br><a href="#examine-panel-chart">Examine panel chart</a>
</td>
</tr>
<tr class="even">
<td>Export</td>
<td align="center"><img src="images/Export.png" width="386"></td>
<td>
<a href="#download-output">Download panel output</a> <br><a href="#extract-r-code">Extract R code</a> <br><a href="#display-panel-settings">Display panel settings</a>
</td>
</tr>
<tr class="odd">
<td>Documentation</td>
<td align="center"><img src="images/Documentation.png" width="337"></td>
<td>
<a href="#quick-tour">Quick tour</a> <br><a href="#browse-vignettes">Open the vignette</a>
</td>
</tr>
<tr class="even">
<td>Additional info</td>
<td align="center"><img src="images/AdditionalInfo.png" width="298"></td>
<td>
<a href="#app-session-info">About this session</a> <br><a href="#about-isee">About iSEE</a>
</td>
</tr>
</tbody>
</table>
<div class="section level3">
<h3 id="organize-panels">Organize panels<a class="anchor" aria-label="anchor" href="#organize-panels"></a>
</h3>
<p>As mentioned above, the default behaviour of the <code><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE()</a></code>
function is to launch an instance of the user interface that displays
one panel of each of the standard types (provided the underlying data is
available, e.g. for reduced dimension plots the <code>reducedDim</code>
slot is required). However, in some cases it is desirable to have
multiple panels of the same type, and/or exclude some panel types.</p>
<p>In order to accommodate such situations, users can add, remove,
change the order of and resize all panels via the
<code>Organization</code> menu in the top-right corner.</p>
<p>Clicking in the selectize box listing all current panels will present
you with a drop-down menu from which you can choose additional panels to
add. Similarly, panels can be removed by clicking on the
<i class="fa fa-times"></i> icon associated with the panel name.</p>
<p>Each panel can be individually resized by changing the width and
height. Note that the total width of a row in the interface is 12 units.
When the width of a panel is greater than the space available, the panel
is moved to a new row.</p>
<p><img src="images/panel-organization.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="examine-panel-chart">Linking panels and transmitting point selections<a class="anchor" aria-label="anchor" href="#examine-panel-chart"></a>
</h3>
<p>When exploring data, it is often useful to be able to select a subset
of points and investigate different aspects of their characteristic
features. In <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>, this
can be achieved by <em>selecting</em> points in one panel, and
<em>transmitting</em> this selection to one or more other panels.</p>
<p>The brushing and point selection can also be programmatically
preconfigured.</p>
<p>The button <code>Examine panel chart</code> display a graph that
reports any active point transmission between panels in the app.</p>
<p><img src="images/panel-graph.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="download-output">Download panel output<a class="anchor" aria-label="anchor" href="#download-output"></a>
</h3>
<p>The <code>Download panel output</code> button opens a modal window
listing all the current panels in the app. Checkboxes allow users to
select any subset of panels to export. Finally, clicking the
<code>Download</code> button in that modal will prompt the app to save
plots to PDF files, tables to CSV files, package the set of files in a
ZIP archive that users can download and save on their computer.</p>
<p><img src="images/panel-download.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="extract-r-code">Extract R code<a class="anchor" aria-label="anchor" href="#extract-r-code"></a>
</h3>
<p>The fact that data exploration is done interactively is no reason to
forego reproducibility! To this end, <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> lets
you export the exact R code used to create each of the currently visible
plots.</p>
<p>Importantly, the script reported by <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>
contains a short preamble needed to set up variable names that are used
in individual panels, including active brushes used to transfer point
selections.</p>
<p><img src="images/code-tracker.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="display-panel-settings">Display panel settings<a class="anchor" aria-label="anchor" href="#display-panel-settings"></a>
</h3>
<p>It can take a great amount of time to achieve a satisfactory panel
configuration. To avoid the need to manually organize the panels each
time the app is opened, <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> offers
the possibility to export code that can be reused later to
programmatically specify how the app is initialized, as well as to
inspect and export the current panel settings for future use.</p>
<p><img src="images/panel-settings.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="quick-tour">Quick tour<a class="anchor" aria-label="anchor" href="#quick-tour"></a>
</h3>
<p>One important aspect of visualization is the ability to share your
insights with others. A powerful way of easily getting people unfamiliar
with your data up to speed is to provide a walkthrough of the interface
and the different types of plots that are displayed. With <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>, this
can be achieved using <em>tours</em>. To configure a tour, you need to
create a text file with two columns; named <code>element</code> and
<code>intro</code>, with the former describing the UI element to
highlight in each step, and the latter providing the descriptive text
that will be displayed.</p>
<p>Clicking the <code>Quick tour</code> button launches an interactive
tour of the interface using the <em><a href="https://CRAN.R-project.org/package=rintrojs" class="external-link">rintrojs</a></em>
package, highlighting specific elements of the user interface, labeled
with information and instructions guiding users through panels and tasks
specific to individual apps.</p>
<p><img src="images/quick-tour.png" width="100%"></p>
<p>You can see a live example of a more complex tour in action in the
deployed apps exemplified on the iSEE website, for example <a href="https://hbcc-nimh.shinyapps.io/shinyApp_JNS2023/" class="external-link uri">https://hbcc-nimh.shinyapps.io/shinyApp_JNS2023/</a>.</p>
</div>
<div class="section level3">
<h3 id="browse-vignettes">Open the vignette<a class="anchor" aria-label="anchor" href="#browse-vignettes"></a>
</h3>
<p>The user interface navigation bar also includes this button to open
the introductory vignette to <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> in your
we browser.</p>
<p>Depending on the version of <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> that
you are using, this will adaptively lead you to a locally built vignette
present on your computer, or the <em>release</em> or the <em>devel</em>
version of the Bioconductor package landing page, e.g. &lt;<a href="https://bioconductor.org/packages/3.21/iSEE/vignettes/basic.html" class="external-link">Bioconductor
3.21</a>&gt;.</p>
</div>
<div class="section level3">
<h3 id="app-session-info">Session information<a class="anchor" aria-label="anchor" href="#app-session-info"></a>
</h3>
<p>This button displays the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code>, which
is a useful piece of information to report when reporting an issue with
an app.</p>
<p><img src="images/session-info.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="about-isee">About iSEE<a class="anchor" aria-label="anchor" href="#about-isee"></a>
</h3>
<p>This button provides information about the authors of <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>, as
well as citation information.</p>
<p>If you use this package, please use the following citation
information:</p>
<blockquote>
<p>Rue-Albrecht K, Marini F, Soneson C, Lun ATL (2018). “iSEE:
Interactive SummarizedExperiment Explorer.” <em>F1000Research</em>,
<em>7</em>, 741. doi: 10.12688/f1000research.14966.1 (URL: <a href="https://doi.org/10.12688/f1000research.14966.1" class="external-link uri">https://doi.org/10.12688/f1000research.14966.1</a>).</p>
</blockquote>
<p>A BibTeX entry for LaTeX users is:</p>
<pre><code>@Article{,
  title = {iSEE: Interactive SummarizedExperiment Explorer},
  author = {Kevin Rue-Albrecht and Federico Marini and Charlotte Soneson and Aaron T. L. Lun},
  publisher = {F1000 Research, Ltd.},
  journal = {F1000Research},
  year = {2018},
  month = {Jun},
  volume = {7},
  pages = {741},
  doi = {10.12688/f1000research.14966.1},
}</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="using-isee-what-can-you-do-with-it">Using iSEE: what can you do with it?<a class="anchor" aria-label="anchor" href="#using-isee-what-can-you-do-with-it"></a>
</h2>
<p>In this section, we list a number of actions you can perform with the
help of iSEE. Feel free to try them out already! The following vignettes
will go into more details for many of them.</p>
<ul>
<li>Organize the iSEE panels:
<ul>
<li>Search for the <code>Organize panels</code> button.</li>
<li>Try to add and remove panels, and resize and reorder the existing
ones.</li>
<li>Remember to click on <code>Apply settings</code> afterwards.</li>
<li>See how to export the <code>initial</code> configuration of
panels.</li>
<li>Relaunch the application, now with the additional argument
<code>initial</code> provided.</li>
</ul>
</li>
<li>Multiple panels of the same type:
<ul>
<li>Show two <em>Reduced dimension plot</em> panels; one showing the PCA
representation, one showing the t-SNE representation.</li>
<li>Link them together, so that points that are selected in one are
highlighted in the other.</li>
</ul>
</li>
<li>More on <code>iSEE</code> links and transmissions:
<ul>
<li>Start an instance with the default set of panels and explore the
different types of plots generated by the Feature assay plot panel using
the choices available for the two axes.</li>
<li>Select a set of points in one of the <em>Feature assay plot</em>
panels, by drawing a rectangle around them.</li>
<li>Then open the <em>Selection parameters</em> collapsible box of
another <em>Feature assay plot</em> panel, and select the <em>Feature
assay plot</em> panel where you made the point selection from the
dropdown menu under <strong>Receive column selection from</strong>.</li>
<li>Note how the points corresponding to the cells that you selected in
the first panel are highlighted in the receiving panel.</li>
<li>You can highlight the points in different ways by changing the
<strong>Selection effect</strong>.</li>
<li>
<code>iSEE</code> can display active links transmitting point
selections between panel using the “<a href="#additional-controls">Examine panel chart</a>” menu.</li>
</ul>
</li>
<li>Color the points:
<ul>
<li>Open the <code>Visual parameters</code> collapsible box in one of
the <em>Reduced dimension plot</em> panels, and set
<code>Color by:</code> to <code>Column data</code>.</li>
<li>Color the cells by <code>Cluster</code>, which contains the cluster
labels that were assigned to the cells in the preprocessing step.</li>
<li>Color the cells by the log10-transformed number of detected genes
(<code>log10_total</code>).</li>
<li>Set <code>Color by:</code> to <code>Feature name</code> in one of
the <em>Reduced dimension plot</em> panels, and select one of the genes
from the dropdown menu. You can search for a gene of interest by typing
in the dropdown box.</li>
</ul>
</li>
<li>Highlight &amp; zoom:
<ul>
<li>Set <code>Color by:</code> to <code>Sample name</code> in one of the
Reduced dimension plots, and use the dropdown menu underneath to change
the highlighted cell.</li>
<li>Reduce the point size and increase the transparency (i.e. reduce the
value for the alpha attribute).</li>
<li>Apply a downsampling grid of 100 horizontal and vertical bins.</li>
<li>As above, select a region in the first panel and transmit the
selection to the second one.</li>
<li>Double-click on the area selected in the first panel to zoom and
display the same view as the second panel.</li>
<li>Double click again anywhere in the panel to zoom out.</li>
</ul>
</li>
<li>Launch iSEE with pre-specified configurations:
<ul>
<li>Start <code>iSEE</code> in the default configuration.</li>
<li>Click on the export icon (<i class="fa fa-download fa-1g"></i>) in
the top-right corner, and select ‘Display panel settings’.</li>
<li>Copy all the code shown in the pop-up window.</li>
<li>Close the app and paste this code in your R session.</li>
<li>This defines the ‘initial’ list that we have composed manually in
the examples above. Then launch a new instance using the following
code.</li>
<li>Note how the app starts in the same configuration as it was before
closing. Of course, it is still possible to continue exploring the data
interactively - we have only changed the starting configuration.</li>
<li>Start <code>iSEE</code> with one <em>Column data plot</em> and one
<em>Row data plot</em> panel.</li>
<li>Start <code>iSEE</code> in an empty configuration.</li>
</ul>
</li>
<li>Export code
<ul>
<li>Click on the export icon (<i class="fa fa-download fa-1g"></i>) in
the top-right corner, and select ‘Extract the R code’.</li>
<li>Copy the preamble and the code for the first plot from the pop-up
window. Close the app and paste the code in your R session.</li>
<li>Note how this recreates the first plot from the app.</li>
</ul>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="modes-and-additional-panels">Modes and additional panels<a class="anchor" aria-label="anchor" href="#modes-and-additional-panels"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/iSEEu" class="external-link">iSEEu</a></em>
(“iSEE universe”) Bioconductor package defines additional custom panels
and predefined ‘modes’ (startup configurations) that may be useful for
specific applications. Here we illustrate the use of the <em>reduced
dimension</em> mode, which will start an application with one reduced
dimension panel for each reduced dimension representation in the input
object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEu" class="external-link">"iSEEu"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: iSEEhex</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEu/man/modeReducedDim.html" class="external-link">modeReducedDim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/02-isee-iseeu-modereduceddim.png" width="100%"></p>
<p>Furthermore, since <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> version
2.0.0, users can leverage the implementation of panels as a hierarchy of
S4 classes to rapidly extend the framework and develop new types of
panels with virtually unlimited freedom of functionality. In this
framework, new panel types can be readily integrated in apps alongside
built-in panel types and immediately benefit of most panel functionality
with minimal effort, including the capacity to transmit selections to
and from other panels. New panels may be implemented by extending the
core virtual class <code>Panel</code> directly for full control over the
user interface and reactive observers; alternatively, it is often
desirable for developers to inherit from one of the concrete panel
classes to get most of the essential functionality from the parent class
“for free”.</p>
<p>Use cases demonstrating the implementation of new panels with various
levels of complexity are available in the <a href="https://isee.github.io/iSEE-book/" class="external-link">Extending iSEE</a> bookdown.
For example, the <em>Hexagonal reduced dimension plot</em> - implemented
in the <em><a href="https://bioconductor.org/packages/3.21/iSEEhex" class="external-link">iSEEhex</a></em>
package - demonstrates an alternative to the downsampling strategy, by
summarizing data points into hexagonal bins.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://isee.github.io/iSEE/reference/ReducedDimensionPlot-class.html" class="external-link">ReducedDimensionPlot</a></span><span class="op">(</span>PanelWidth <span class="op">=</span> <span class="fl">6L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/iSEEhex/man/ReducedDimensionHexPlot-class.html" class="external-link">ReducedDimensionHexPlot</a></span><span class="op">(</span>PanelWidth <span class="op">=</span> <span class="fl">6L</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/02-isee-iseeu-modereduceddimhex.png" width="100%"></p>
<p>Another useful mode you can use with your data, for example if using
iSEE for mass cytometry data, would be the <code>modeGating</code>. This
launches an app preconfigured with multiple chain-linked feature
expression plots for interactive data exploration.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Select top variable genes ----</span></span>
<span><span class="va">plot_count</span> <span class="op">&lt;-</span> <span class="fl">6</span></span>
<span><span class="va">rv</span> <span class="op">&lt;-</span> <span class="fu">rowVars</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"logcounts"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">top_var</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">rv</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>, <span class="va">plot_count</span><span class="op">*</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">top_var_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="va">top_var</span><span class="op">]</span></span>
<span></span>
<span><span class="va">plot_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">top_var_genes</span>, <span class="va">plot_count</span><span class="op">)</span>,</span>
<span>  y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">top_var_genes</span>, <span class="va">plot_count</span><span class="op">)</span>,</span>
<span>  stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># launch the app itself ----</span></span>
<span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEu/man/modeGating.html" class="external-link">modeGating</a></span><span class="op">(</span><span class="va">sce</span>, </span>
<span>                  plotAssay <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>                  features <span class="op">=</span> <span class="va">plot_features</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/02-isee-iseeu-modegating.png" width="100%"></p>
<p>Interestingly, iSEEu also has a set of extra panels that can come in
handy in different situations. For example,
<code><a href="https://rdrr.io/pkg/iSEEu/man/AggregatedDotPlot.html" class="external-link">iSEEu::AggregatedDotPlot</a></code> implements an aggregated dot plot
where each feature/group combination is represented by a dot. The color
of the dot scales with the mean assay value across all samples for a
given group, while the size of the dot scales with the proportion of
non-zero values across samples in that group. This can be an alternative
to a heatmap, and an example can be seen in the chunks below:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span></span>
<span>  <span class="va">sce</span>, </span>
<span>  initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEu/man/AggregatedDotPlot.html" class="external-link">AggregatedDotPlot</a></span><span class="op">(</span></span>
<span>      Assay <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>      ColumnDataLabel <span class="op">=</span> <span class="st">"labels_main"</span>,</span>
<span>      CustomRowsText <span class="op">=</span> <span class="va">top_var_genes</span></span>
<span>    <span class="op">)</span></span>
<span>    </span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images/02-isee-iseeu-aggrodotplot.png" width="100%"></p>
<p>An overview of other extension packages and the panels they provide
can be found on the iSEE website: <a href="https://isee.github.io/panels.html" class="external-link uri">https://isee.github.io/panels.html</a>.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><summary><b> Session info </b>
</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] iSEEu_1.20.0                iSEEhex_1.10.0             </span></span>
<span><span class="co">#&gt;  [3] iSEE_2.20.0                 SingleCellExperiment_1.30.1</span></span>
<span><span class="co">#&gt;  [5] SummarizedExperiment_1.38.1 Biobase_2.68.0             </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.60.0        GenomeInfoDb_1.44.0        </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.42.0              S4Vectors_0.46.0           </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.54.0         generics_0.1.4             </span></span>
<span><span class="co">#&gt; [13] MatrixGenerics_1.20.0       matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [15] BiocStyle_2.36.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] rlang_1.1.6              magrittr_2.0.3           shinydashboard_0.7.3    </span></span>
<span><span class="co">#&gt;  [4] clue_0.3-66              GetoptLong_1.0.5         compiler_4.5.1          </span></span>
<span><span class="co">#&gt;  [7] mgcv_1.9-3               png_0.1-8                systemfonts_1.2.3       </span></span>
<span><span class="co">#&gt; [10] vctrs_0.6.5              pkgconfig_2.0.3          shape_1.4.6.1           </span></span>
<span><span class="co">#&gt; [13] crayon_1.5.3             fastmap_1.2.0            XVector_0.48.0          </span></span>
<span><span class="co">#&gt; [16] fontawesome_0.5.3        promises_1.3.3           rmarkdown_2.29          </span></span>
<span><span class="co">#&gt; [19] UCSC.utils_1.4.0         shinyAce_0.4.4           ragg_1.4.0              </span></span>
<span><span class="co">#&gt; [22] xfun_0.52                cachem_1.1.0             jsonlite_2.0.0          </span></span>
<span><span class="co">#&gt; [25] listviewer_4.0.0         later_1.4.2              DelayedArray_0.34.1     </span></span>
<span><span class="co">#&gt; [28] parallel_4.5.1           cluster_2.1.8.1          R6_2.6.1                </span></span>
<span><span class="co">#&gt; [31] bslib_0.9.0              RColorBrewer_1.1-3       jquerylib_0.1.4         </span></span>
<span><span class="co">#&gt; [34] Rcpp_1.1.0               iterators_1.0.14         knitr_1.50              </span></span>
<span><span class="co">#&gt; [37] httpuv_1.6.16            Matrix_1.7-3             splines_4.5.1           </span></span>
<span><span class="co">#&gt; [40] igraph_2.1.4             tidyselect_1.2.1         abind_1.4-8             </span></span>
<span><span class="co">#&gt; [43] yaml_2.3.10              doParallel_1.0.17        codetools_0.2-20        </span></span>
<span><span class="co">#&gt; [46] miniUI_0.1.2             lattice_0.22-7           tibble_3.3.0            </span></span>
<span><span class="co">#&gt; [49] shiny_1.11.1             evaluate_1.0.4           desc_1.4.3              </span></span>
<span><span class="co">#&gt; [52] circlize_0.4.16          pillar_1.11.0            BiocManager_1.30.26     </span></span>
<span><span class="co">#&gt; [55] DT_0.33                  foreach_1.5.2            shinyjs_2.1.0           </span></span>
<span><span class="co">#&gt; [58] ggplot2_3.5.2            sparseMatrixStats_1.20.0 scales_1.4.0            </span></span>
<span><span class="co">#&gt; [61] xtable_1.8-4             glue_1.8.0               tools_4.5.1             </span></span>
<span><span class="co">#&gt; [64] hexbin_1.28.5            colourpicker_1.3.0       fs_1.6.6                </span></span>
<span><span class="co">#&gt; [67] grid_4.5.1               colorspace_2.1-1         nlme_3.1-168            </span></span>
<span><span class="co">#&gt; [70] GenomeInfoDbData_1.2.14  vipor_0.4.7              cli_3.6.5               </span></span>
<span><span class="co">#&gt; [73] textshaping_1.0.1        S4Arrays_1.8.1           viridisLite_0.4.2       </span></span>
<span><span class="co">#&gt; [76] ComplexHeatmap_2.24.1    dplyr_1.1.4              gtable_0.3.6            </span></span>
<span><span class="co">#&gt; [79] rintrojs_0.3.4           sass_0.4.10              digest_0.6.37           </span></span>
<span><span class="co">#&gt; [82] SparseArray_1.8.0        ggrepel_0.9.6            rjson_0.2.23            </span></span>
<span><span class="co">#&gt; [85] htmlwidgets_1.6.4        farver_2.1.2             memoise_2.0.1           </span></span>
<span><span class="co">#&gt; [88] htmltools_0.5.8.1        pkgdown_2.1.3            lifecycle_1.0.4         </span></span>
<span><span class="co">#&gt; [91] httr_1.4.7               shinyWidgets_0.9.0       GlobalOptions_0.1.2     </span></span>
<span><span class="co">#&gt; [94] mime_0.13</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Bremner1994-hintonplots" class="csl-entry">
Bremner, Frederick J, Stephen J Gotts, and Dina L Denham. 1994.
<span>“Hinton Diagrams: Viewing Connection Strengths in Neural
Networks.”</span> <em>Behav. Res. Methods Instrum. Comput.</em> 26 (2):
215–18.
</div>
<div id="ref-Hinton1991-hintondiagram" class="csl-entry">
Hinton, Geoffrey E, and Tim Shallice. 1991. <span>“Lesioning an
Attractor Network: Investigations of Acquired Dyslexia.”</span>
<em>Psychological Review</em> 98 (1): 74–95.
</div>
<div id="ref-Rue-Albrecht2018-isee" class="csl-entry">
Rue-Albrecht, Kevin, Federico Marini, Charlotte Soneson, and Aaron T L
Lun. 2018. <span>“<span class="nocase">iSEE</span>: Interactive
<span>SummarizedExperiment</span> Explorer.”</span> <em>F1000Res.</em>
7: 741.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Federico Marini, Kevin Rue-Albrecht, Charlotte Soneson, Aaron Lun, Najla Abassi.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
