<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Bonus content • iUSEiSEE</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Bonus content">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iUSEiSEE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Workshop sessions

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/d00_about_this_workshop.html">00. About this workshop</a>
    </li>
    <li>
      <a href="../articles/d01_preparation.html">01. Preparation</a>
    </li>
    <li>
      <a href="../articles/d02_iSEE_overview.html">02. An overview of iSEE</a>
    </li>
    <li>
      <a href="../articles/d03_iSEE_quickrecipes.html">03. The iSEE cookbook</a>
    </li>
    <li>
      <a href="../articles/d04_bonus_content.html">04. Bonus iSEE content</a>
    </li>
    <li>
      <a href="../articles/d05_extending_iSEE.html">05. Extending iSEE</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/iSEE/iUSEiSEE" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><link href="d04_bonus_content_files/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="d04_bonus_content_files/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Bonus content</h1>
                        <h4 data-toc-skip class="author">Federico
Marini<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>,
Kevin Rue-Albrecht<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, Charlotte soneson<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>, Aaron Lun<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>, Najla
Abassi<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iUSEiSEE/blob/update-2025/vignettes/d04_bonus_content.Rmd" class="external-link"><code>vignettes/d04_bonus_content.Rmd</code></a></small>
      <div class="hidden name"><code>d04_bonus_content.Rmd</code></div>

    </div>

    
    
<p><img src="iSEE.png" align="right" alt="" width="120"></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette contains some additional examples of things that can be
done using <code>iSEE</code>, going beyond the basic application
interface.</p>
</div>
<div class="section level2">
<h2 id="iseede-and-iseepathways">iSEEde and iSEEpathways<a class="anchor" aria-label="anchor" href="#iseede-and-iseepathways"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.21/iSEEde" class="external-link">iSEEde</a></em> and
<em><a href="https://bioconductor.org/packages/3.21/iSEEpathways" class="external-link">iSEEpathways</a></em>
and ideal companion packages for exploring differential expression
results in (e.g.) bulk RNA-seq data.</p>
<p>We will use the <code>macrophage</code> dataset (derived from the
work of <span class="citation">(Alasoo et al. 2018)</span>), for which
we have already run a differential expression analysis using
<code>DESeq2</code> <span class="citation">(Love, Huber, and Anders
2014)</span>. The processed dataset is provided with the workshop
package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">macrophage_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"datasets"</span>, <span class="st">"sce_macrophage_readytouse.RDS"</span>,</span>
<span>                                   package <span class="op">=</span> <span class="st">"iUSEiSEE"</span><span class="op">)</span></span>
<span><span class="va">macrophage_location</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmp7gC9Da/temp_libpath66ef12b04acf/iUSEiSEE/datasets/sce_macrophage_readytouse.RDS"</span></span>
<span></span>
<span><span class="va">sce_macrophage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">macrophage_location</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://isee.github.io/iSEE/" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEde" class="external-link">"iSEEde"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEpathways" class="external-link">"iSEEpathways"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/AnnotationDbi" class="external-link">"AnnotationDbi"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"GO.db"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p><em><a href="https://bioconductor.org/packages/3.21/iSEEde" class="external-link">iSEEde</a></em> and
<em><a href="https://bioconductor.org/packages/3.21/iSEEpathways" class="external-link">iSEEpathways</a></em>
are two new Bioconductor packages that provide <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> panels
specifically aimed towards exploration of differential expression and
pathway analysis results. More precisely, <em><a href="https://bioconductor.org/packages/3.21/iSEEde" class="external-link">iSEEde</a></em>
provides the <code>VolcanoPlot</code>, <code>MAPlot</code>,
<code>LogFCLogFCPlot</code> and <code>DETable</code> panels. These
panels can be configured to extract data that was added via the
<code><a href="https://rdrr.io/pkg/iSEEde/man/utils-SummarizedExperiment.html" class="external-link">embedContrastResults()</a></code> function above. Let’s look at an
example:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce_macrophage</span>, initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu">iSEEde</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/DETable-class.html" class="external-link">DETable</a></span><span class="op">(</span></span>
<span>    ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span>,</span>
<span>    HiddenColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"baseMean"</span>, <span class="st">"lfcSE"</span>, <span class="st">"stat"</span><span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span>  <span class="fu">iSEEde</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/VolcanoPlot-class.html" class="external-link">VolcanoPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span>,</span>
<span>  <span class="fu">iSEEde</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/MAPlot-class.html" class="external-link">MAPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/04-isee-de-setup1.png" width="992"></p>
<!-- Possible actions: -->
<!-- select some genes from any panel and pass the selection to others -->
<!-- move panels around, add some, update configuration -->
<!-- hover with the mouse to have the tooltip show up -->
<!-- usual iSEE magic: export code for the plots -->
<p>Note how it is easy to switch to a different contrast in any of the
panels.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce_macrophage</span>, initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu">iSEEde</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/DETable-class.html" class="external-link">DETable</a></span><span class="op">(</span></span>
<span>    ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span>,</span>
<span>    HiddenColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"baseMean"</span>, <span class="st">"lfcSE"</span>, <span class="st">"stat"</span><span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span>  <span class="fu">iSEEde</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/VolcanoPlot-class.html" class="external-link">VolcanoPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span>,</span>
<span>  <span class="fu">iSEEde</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/MAPlot-class.html" class="external-link">MAPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/iSEEpathways/man/PathwaysTable-class.html" class="external-link">PathwaysTable</a></span><span class="op">(</span></span>
<span>    ResultName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.limma.fgsea"</span>,</span>
<span>    Selected <span class="op">=</span> <span class="st">"GO:0046324"</span></span>
<span>  <span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://isee.github.io/iSEE/reference/ComplexHeatmapPlot-class.html" class="external-link">ComplexHeatmapPlot</a></span><span class="op">(</span></span>
<span>    RowSelectionSource <span class="op">=</span> <span class="st">"PathwaysTable1"</span>,</span>
<span>    CustomRows <span class="op">=</span> <span class="cn">FALSE</span>, ColumnData <span class="op">=</span> <span class="st">"condition_name"</span>,</span>
<span>    ClusterRows <span class="op">=</span> <span class="cn">TRUE</span>, Assay <span class="op">=</span> <span class="st">"vst"</span></span>
<span>  <span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/iSEEpathways/man/FgseaEnrichmentPlot-class.html" class="external-link">FgseaEnrichmentPlot</a></span><span class="op">(</span></span>
<span>    ResultName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.limma.fgsea"</span>,</span>
<span>    PathwayId <span class="op">=</span> <span class="st">"GO:0046324"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/04-isee-de-setup2.png" width="992"></p>
<!-- Possible actions: -->
<!-- select a pathway from PAT1 -->
<!-- transform vst data in heatmap, center -->
<!-- receive row selection in volcano and MA plot -->
<!-- change coloring to "row selection" -->
<!-- select alternative contrast and see all dependencies updated -->
<!-- usual iSEE magic: export code for the plots -->
<!-- usual iSEE magic: export code for the panel configuration -->
</div>
<div class="section level2">
<h2 id="iseefier">iSEEfier<a class="anchor" aria-label="anchor" href="#iseefier"></a>
</h2>
<p>Let’s say we are interested in visualizing the expression of a list
of specific marker genes in one view, or maybe we created different
initial states separately, but would like to visualize them in the same
instance. As we previously learned, we can do a lot of these tasks by
running the command:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># don't run</span></span>
<span><span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<p>Then, add/remove the panels to our need. This can involve multiple
steps (selecting the gene of interest, color by a specific
<code>colData</code>…), or probably even write multiple lines of code.
For that we can use the <em><a href="https://bioconductor.org/packages/3.21/iSEEfier" class="external-link">iSEEfier</a></em>
package, which streamlines the process of starting (or if you will,
firing up) an <code>iSEE</code> instance with a small chunk of code,
avoiding the tedious way of setting up every <code>iSEE</code> panel
individually.</p>
<p>In this section, we will illustrate a simple example of how to use
<em><a href="https://bioconductor.org/packages/3.21/iSEEfier" class="external-link">iSEEfier</a></em>.
We will use the same <code>pbmc3k</code> data we worked with during this
workshop.</p>
<p>We start by loading the data:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/NajlaAbassi/iSEEfier" class="external-link">"iSEEfier"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># import data</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"datasets"</span>, <span class="st">"sce_pbmc3k.RDS"</span>, package <span class="op">=</span> <span class="st">"iUSEiSEE"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 32738 2643 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span></span>
<span><span class="co">#&gt; rowData names(19): ENSEMBL_ID Symbol_TENx ... FDR_cluster11</span></span>
<span><span class="co">#&gt;   FDR_cluster12</span></span>
<span><span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span></span>
<span><span class="co">#&gt; colData names(24): Sample Barcode ... labels_ont cell_ontology_labels</span></span>
<span><span class="co">#&gt; reducedDimNames(3): PCA TSNE UMAP</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>For example, we can be interested in visualizing the expression of
<em>GZMB</em>, <em>TGFB</em>, and <em>CD28</em> genes all at once. We
start by providing a couple of parameters:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># define the list of genes</span></span>
<span><span class="va">feature_list_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GZMB"</span>, <span class="st">"TGFB1"</span>, <span class="st">"CD28"</span><span class="op">)</span></span>
<span><span class="co"># define the cluster/cell type </span></span>
<span><span class="va">cluster_1</span> <span class="op">&lt;-</span> <span class="st">"labels_main"</span></span></code></pre></div>
<p>Now we can pass these parameters into <code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEinit.html" class="external-link">iSEEinit()</a></code> to
create a customized initial configuration:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create an initial state with iSEEinit</span></span>
<span><span class="va">initial_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEinit.html" class="external-link">iSEEinit</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>                      features <span class="op">=</span> <span class="va">feature_list_1</span>,</span>
<span>                      clusters <span class="op">=</span> <span class="va">cluster_1</span>,</span>
<span>                      add_markdown_panel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>The rest can be as easy as passing this initial to the
<code><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE()</a></code> call:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial <span class="op">=</span> <span class="va">initial_1</span><span class="op">)</span></span></code></pre></div>
<p>This is how it would look like:</p>
<p><img src="images/04-isee-on-fier.png" width="100%"></p>
<p>While we are visualizing the expression of these genes, we might want
to take some notes (gene X is more expressed in a certain cell
type/cluster than some others, maybe we are trying to annotate the cells
ourselves if the annotation wasn’t available…).<br>
For this, we used the argument <code>add_markdown_panel = TRUE</code>.
It will display a <code>MarkdownBoard</code> panel where we can note our
observations without leaving the app.</p>
<p>We can check the initial’s content, or how the included panels are
linked between each other without running the app with
<code><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_tiles.html" class="external-link">view_initial_tiles()</a></code> and
<code><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_network.html" class="external-link">view_initial_network()</a></code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># display a graphical representation of the initial configuration, where the panels are identified by their corresponding colors</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_tiles.html" class="external-link">view_initial_tiles</a></span><span class="op">(</span>initial <span class="op">=</span> <span class="va">initial_1</span><span class="op">)</span></span></code></pre></div>
<p><img src="d04_bonus_content_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># display a network visualization for the panels</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_network.html" class="external-link">view_initial_network</a></span><span class="op">(</span><span class="va">initial_1</span>, plot_format <span class="op">=</span> <span class="st">"igraph"</span><span class="op">)</span></span></code></pre></div>
<p><img src="d04_bonus_content_files/figure-html/unnamed-chunk-13-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; IGRAPH b7bd62a DN-- 14 4 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), color (v/c)</span></span>
<span><span class="co">#&gt; + edges from b7bd62a (vertex names):</span></span>
<span><span class="co">#&gt; [1] ReducedDimensionPlot1-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [2] ReducedDimensionPlot2-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [3] ReducedDimensionPlot3-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [4] ReducedDimensionPlot4-&gt;FeatureAssayPlot4</span></span></code></pre>
<p>Another alternative for network visualization would use the
interactive widget provided by <code>visNetwork</code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_network.html" class="external-link">view_initial_network</a></span><span class="op">(</span><span class="va">initial_1</span>, plot_format <span class="op">=</span> <span class="st">"visNetwork"</span><span class="op">)</span></span>
<span><span class="co">#&gt; IGRAPH 04159f5 DN-- 14 4 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), color (v/c)</span></span>
<span><span class="co">#&gt; + edges from 04159f5 (vertex names):</span></span>
<span><span class="co">#&gt; [1] ReducedDimensionPlot1-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [2] ReducedDimensionPlot2-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [3] ReducedDimensionPlot3-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [4] ReducedDimensionPlot4-&gt;FeatureAssayPlot4</span></span></code></pre></div>
<p>It is also possible to combine multiple initials into one:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feature_list_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD74"</span>, <span class="st">"CD79B"</span><span class="op">)</span></span>
<span><span class="va">initial_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEinit.html" class="external-link">iSEEinit</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>                      features <span class="op">=</span> <span class="va">feature_list_2</span>,</span>
<span>                      clusters <span class="op">=</span> <span class="va">cluster_1</span><span class="op">)</span></span>
<span><span class="va">merged_config</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/glue_initials.html" class="external-link">glue_initials</a></span><span class="op">(</span><span class="va">initial_1</span>, <span class="va">initial_2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Merging together 2 `initial` configuration objects...</span></span>
<span><span class="co">#&gt; Combining sets of 14, 10 different panels.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Dropping 1 of the original list of 24 (detected as duplicated entries)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Some names of the panels were specified by the same name, but this situation can be handled at runtime by iSEE</span></span>
<span><span class="co">#&gt; (This is just a non-critical message)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Returning an `initial` configuration including 23 different panels. Enjoy!</span></span>
<span><span class="co">#&gt; If you want to obtain a preview of the panels configuration, you can call `view_initial_tiles()` on the output of this function</span></span>
<span><span class="co"># check out the content of merged_config</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_tiles.html" class="external-link">view_initial_tiles</a></span><span class="op">(</span>initial <span class="op">=</span> <span class="va">merged_config</span><span class="op">)</span></span></code></pre></div>
<p><img src="d04_bonus_content_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p><code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEfier-pkg.html" class="external-link">?iSEEfier</a></code> is always your friend whenever you need
further documentation on the package/a certain function and how to use
it.</p>
</div>
<div class="section level2">
<h2 id="iseehub-iseeindex">iSEEhub &amp; iSEEindex<a class="anchor" aria-label="anchor" href="#iseehub-iseeindex"></a>
</h2>
<div class="section level3">
<h3 id="iseehub-iseeing-the-experimenthub-datasets">iSEEhub: <code>iSEE</code>ing the ExperimentHub datasets<a class="anchor" aria-label="anchor" href="#iseehub-iseeing-the-experimenthub-datasets"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.21/iSEEhub" class="external-link">iSEEhub</a></em>
package provides a custom landing page for an <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em>
application interfacing with the Bioconductor <em><a href="https://bioconductor.org/packages/3.21/ExperimentHub" class="external-link">ExperimentHub</a></em>.
The landing page allows users to browse the ExperimentHub, select a data
set, download and cache it, and import it directly into an iSEE app.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://isee.github.io/iSEE/" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEhub" class="external-link">"iSEEhub"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: ExperimentHub</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationHub</span></span>
<span><span class="co">#&gt; Loading required package: BiocFileCache</span></span>
<span><span class="co">#&gt; Loading required package: dbplyr</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'AnnotationHub'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     cache</span></span>
<span></span>
<span><span class="va">ehub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEhub/man/iSEEhub.html" class="external-link">iSEEhub</a></span><span class="op">(</span><span class="va">ehub</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/04-iseehub.png" width="992"></p>
<!-- Possible actions: -->
<!-- select a dataset from DuoClustering -->
<!-- start iSEE on that -->
</div>
<div class="section level3">
<h3 id="iseeindex-one-instance-of-isee-to-explore-them-all">iSEEindex: one instance of <code>iSEE</code> to explore them
all<a class="anchor" aria-label="anchor" href="#iseeindex-one-instance-of-isee-to-explore-them-all"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.21/iSEEindex" class="external-link">iSEEindex</a></em>
provides an interface to <strong>any</strong> collection of data sets
within a single iSEE web-application.<br>
The main functionality of this package is to define a custom landing
page allowing app maintainers to list a custom collection of data sets
that users can select from and directly load objects into an iSEE web
application. To see how to configure such an app, we will create a small
example:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://isee.github.io/iSEE/" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEEindex" class="external-link">"iSEEindex"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="op">(</span>cache <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dataset_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">yaml</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yaml/man/read_yaml.html" class="external-link">read_yaml</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"iSEEindex"</span>, <span class="st">"example.yaml"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">x</span><span class="op">$</span><span class="va">datasets</span></span>
<span><span class="op">}</span></span>
<span><span class="va">initial_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">yaml</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yaml/man/read_yaml.html" class="external-link">read_yaml</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"iSEEindex"</span>, <span class="st">"example.yaml"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">x</span><span class="op">$</span><span class="va">initial</span></span>
<span><span class="op">}</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEindex/man/iSEEindex.html" class="external-link">iSEEindex</a></span><span class="op">(</span><span class="va">bfc</span>, <span class="va">dataset_fun</span>, <span class="va">initial_fun</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/04-iseeindex.png" width="992"></p>
<p>A more elaborate example (referring to the work in <span class="citation">(Rigby et al. 2023)</span>) is available at <a href="https://rehwinkellab.shinyapps.io/ifnresource/" class="external-link">https://rehwinkellab.shinyapps.io/ifnresource/</a>.
The source can be found at <a href="https://github.com/kevinrue/IFNresource" class="external-link">https://github.com/kevinrue/IFNresource</a>.</p>
<!-- Possible actions: -->
<!-- navigate to github repo -->
<!-- explore the app.R file -->
<!-- check the initial state files -->
<!-- have a look at the configuration yaml -->
<p>Potential use cases can include:</p>
<ul>
<li>An app to present and explore the different datasets in your next
publication</li>
<li>An app to explore collection of datasets collaboratively, in
consortium-like initiatives</li>
<li>An app to mirror and enhance the content of e.g. the cellxgene data
portal</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="tours-help-and-storytelling">Tours: help and storytelling<a class="anchor" aria-label="anchor" href="#tours-help-and-storytelling"></a>
</h2>
<p>Tours can be an essential tool to satisfy two needs:</p>
<ul>
<li>Helping users to navigate the UI</li>
<li>Telling a story on an existing configuration</li>
</ul>
<p>A simple example can be demonstrated with this configuration:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"datasets"</span>, <span class="st">"sce_pbmc3k.RDS"</span>, package <span class="op">=</span> <span class="st">"iUSEiSEE"</span><span class="op">)</span></span>
<span><span class="va">sce_location</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmp7gC9Da/temp_libpath66ef12b04acf/iUSEiSEE/datasets/sce_pbmc3k.RDS"</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">sce_location</span><span class="op">)</span></span>
<span></span>
<span><span class="va">initial_for_tour</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://isee.github.io/iSEE/reference/ReducedDimensionPlot-class.html" class="external-link">ReducedDimensionPlot</a></span><span class="op">(</span>PanelWidth <span class="op">=</span> <span class="fl">3L</span><span class="op">)</span>, </span>
<span>  <span class="fu"><a href="https://isee.github.io/iSEE/reference/RowDataTable-class.html" class="external-link">RowDataTable</a></span><span class="op">(</span>PanelWidth <span class="op">=</span> <span class="fl">3L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://isee.github.io/iSEE/reference/FeatureAssayPlot-class.html" class="external-link">FeatureAssayPlot</a></span><span class="op">(</span>PanelWidth <span class="op">=</span> <span class="fl">3L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://isee.github.io/iSEE/reference/ComplexHeatmapPlot-class.html" class="external-link">ComplexHeatmapPlot</a></span><span class="op">(</span>PanelWidth <span class="op">=</span> <span class="fl">3L</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>This next chunk defines the <strong>steps</strong> of the tour,
specified by an anchoring point (<code>element</code>) and the content
of that step (<code>intro</code>).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tour</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  element <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"#Welcome"</span>,</span>
<span>    <span class="st">"#ReducedDimensionPlot1"</span>,</span>
<span>    <span class="st">"#RowDataTable1"</span>,</span>
<span>    <span class="st">"#ComplexHeatmapPlot1"</span>,</span>
<span>    <span class="st">"#FeatureAssayPlot1"</span>,</span>
<span>    <span class="st">"#ReducedDimensionPlot1"</span>,</span>
<span>    <span class="st">"#Conclusion"</span><span class="op">)</span>,</span>
<span>  intro <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Welcome to this tour!"</span>,</span>
<span>    <span class="st">"This is the a reduced dimension plot"</span>,</span>
<span>    <span class="st">"and this is a table"</span>,</span>
<span>    <span class="st">"Why not a heatmap?"</span>,</span>
<span>    <span class="st">"... and now we look at one individual feature."</span>,</span>
<span>    <span class="st">"Back to the a reduced dimension plot..."</span>,</span>
<span>    <span class="st">"Thank you for taking this tour!"</span><span class="op">)</span>,</span>
<span>  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial <span class="op">=</span> <span class="va">initial_for_tour</span>, tour <span class="op">=</span> <span class="va">tour</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/04-isee-simpletour.png" width="992"></p>
</div>
<div class="section level2">
<h2 id="interoperability">Interoperability<a class="anchor" aria-label="anchor" href="#interoperability"></a>
</h2>
<p>As we have seen throughout this workshop, <em><a href="https://bioconductor.org/packages/3.21/iSEE" class="external-link">iSEE</a></em> works
on <code>SummarizedExperiment</code> objects, or derivatives thereof.
However, especially if you are working with single-cell data you may
already have your dataset available in another format (such as a
<code>Seurat</code> object or an <code>AnnData</code> object). To use
<code>iSEE</code> on these objects, they first have to be converted to
<code>SingleCellExperiment</code> objects. Fortunately, there are good
packages available to perform this conversion. We exemplify this below,
using data downloaded from <a href="https://zenodo.org/records/10084595" class="external-link uri">https://zenodo.org/records/10084595</a> (if you want to
follow along, download the data and unpack in a folder named
<code>datasets</code> in the current working directory).</p>
<div class="section level3">
<h3 id="from-seurat">From Seurat<a class="anchor" aria-label="anchor" href="#from-seurat"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://satijalab.org/seurat" class="external-link">"Seurat"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"datasets/seurat_pbmc3k.RDS"</span><span class="op">)</span></span>
<span><span class="va">sce_from_seurat</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/as.SingleCellExperiment.html" class="external-link">as.SingleCellExperiment</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_object</span></span>
<span><span class="op">)</span></span>
<span><span class="va">sce_from_seurat</span></span>
<span></span>
<span><span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce_from_seurat</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="from-anndata">From AnnData<a class="anchor" aria-label="anchor" href="#from-anndata"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/theislab/zellkonverter" class="external-link">"zellkonverter"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce_from_anndata</span> <span class="op">&lt;-</span> <span class="fu">zellkonverter</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/zellkonverter/man/readH5AD.html" class="external-link">readH5AD</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"datasets/anndata_pbmc3k.h5ad"</span><span class="op">)</span></span>
<span><span class="va">sce_from_anndata</span></span>
<span></span>
<span><span class="fu"><a href="https://isee.github.io/iSEE/reference/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce_from_anndata</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><summary><b> Session info </b>
</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] iSEEindex_1.6.0             iSEEhub_1.10.0             </span></span>
<span><span class="co">#&gt;  [3] ExperimentHub_2.16.0        AnnotationHub_3.16.0       </span></span>
<span><span class="co">#&gt;  [5] BiocFileCache_2.16.0        dbplyr_2.5.0               </span></span>
<span><span class="co">#&gt;  [7] iSEEfier_1.4.0              GO.db_3.21.0               </span></span>
<span><span class="co">#&gt;  [9] org.Hs.eg.db_3.21.0         AnnotationDbi_1.70.0       </span></span>
<span><span class="co">#&gt; [11] iSEEpathways_1.6.0          iSEEde_1.6.0               </span></span>
<span><span class="co">#&gt; [13] iSEE_2.20.0                 SingleCellExperiment_1.30.1</span></span>
<span><span class="co">#&gt; [15] SummarizedExperiment_1.38.1 Biobase_2.68.0             </span></span>
<span><span class="co">#&gt; [17] GenomicRanges_1.60.0        GenomeInfoDb_1.44.0        </span></span>
<span><span class="co">#&gt; [19] IRanges_2.42.0              S4Vectors_0.46.0           </span></span>
<span><span class="co">#&gt; [21] BiocGenerics_0.54.0         generics_0.1.4             </span></span>
<span><span class="co">#&gt; [23] MatrixGenerics_1.20.0       matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [25] BiocStyle_2.36.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3      jsonlite_2.0.0          shape_1.4.6.1          </span></span>
<span><span class="co">#&gt;   [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.29         </span></span>
<span><span class="co">#&gt;   [7] GlobalOptions_0.1.2     fs_1.6.6                ragg_1.4.0             </span></span>
<span><span class="co">#&gt;  [10] vctrs_0.6.5             memoise_2.0.1           htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt;  [13] S4Arrays_1.8.1          BiocBaseUtils_1.10.0    curl_6.4.0             </span></span>
<span><span class="co">#&gt;  [16] SparseArray_1.8.0       sass_0.4.10             bslib_0.9.0            </span></span>
<span><span class="co">#&gt;  [19] htmlwidgets_1.6.4       desc_1.4.3              fontawesome_0.5.3      </span></span>
<span><span class="co">#&gt;  [22] httr2_1.2.0             listviewer_4.0.0        cachem_1.1.0           </span></span>
<span><span class="co">#&gt;  [25] igraph_2.1.4            mime_0.13               lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt;  [28] iterators_1.0.14        pkgconfig_2.0.3         colourpicker_1.3.0     </span></span>
<span><span class="co">#&gt;  [31] Matrix_1.7-3            R6_2.6.1                fastmap_1.2.0          </span></span>
<span><span class="co">#&gt;  [34] GenomeInfoDbData_1.2.14 shiny_1.11.1            clue_0.3-66            </span></span>
<span><span class="co">#&gt;  [37] digest_0.6.37           colorspace_2.1-1        paws.storage_0.9.0     </span></span>
<span><span class="co">#&gt;  [40] DESeq2_1.48.1           textshaping_1.0.1       RSQLite_2.4.2          </span></span>
<span><span class="co">#&gt;  [43] filelock_1.0.3          urltools_1.7.3.1        httr_1.4.7             </span></span>
<span><span class="co">#&gt;  [46] abind_1.4-8             mgcv_1.9-3              compiler_4.5.1         </span></span>
<span><span class="co">#&gt;  [49] bit64_4.6.0-1           withr_3.0.2             doParallel_1.0.17      </span></span>
<span><span class="co">#&gt;  [52] BiocParallel_1.42.1     DBI_1.2.3               shinyAce_0.4.4         </span></span>
<span><span class="co">#&gt;  [55] hexbin_1.28.5           rappdirs_0.3.3          DelayedArray_0.34.1    </span></span>
<span><span class="co">#&gt;  [58] rjson_0.2.23            tools_4.5.1             vipor_0.4.7            </span></span>
<span><span class="co">#&gt;  [61] httpuv_1.6.16           glue_1.8.0              nlme_3.1-168           </span></span>
<span><span class="co">#&gt;  [64] promises_1.3.3          grid_4.5.1              cluster_2.1.8.1        </span></span>
<span><span class="co">#&gt;  [67] gtable_0.3.6            XVector_0.48.0          stringr_1.5.1          </span></span>
<span><span class="co">#&gt;  [70] BiocVersion_3.21.1      ggrepel_0.9.6           foreach_1.5.2          </span></span>
<span><span class="co">#&gt;  [73] pillar_1.11.0           limma_3.64.1            later_1.4.2            </span></span>
<span><span class="co">#&gt;  [76] rintrojs_0.3.4          circlize_0.4.16         splines_4.5.1          </span></span>
<span><span class="co">#&gt;  [79] dplyr_1.1.4             lattice_0.22-7          bit_4.6.0              </span></span>
<span><span class="co">#&gt;  [82] paws.common_0.8.4       tidyselect_1.2.1        ComplexHeatmap_2.24.1  </span></span>
<span><span class="co">#&gt;  [85] locfit_1.5-9.12         Biostrings_2.76.0       miniUI_0.1.2           </span></span>
<span><span class="co">#&gt;  [88] knitr_1.50              edgeR_4.6.3             xfun_0.52              </span></span>
<span><span class="co">#&gt;  [91] shinydashboard_0.7.3    statmod_1.5.0           iSEEhex_1.10.0         </span></span>
<span><span class="co">#&gt;  [94] DT_0.33                 stringi_1.8.7           visNetwork_2.1.2       </span></span>
<span><span class="co">#&gt;  [97] UCSC.utils_1.4.0        yaml_2.3.10             shinyWidgets_0.9.0     </span></span>
<span><span class="co">#&gt; [100] evaluate_1.0.4          codetools_0.2-20        tibble_3.3.0           </span></span>
<span><span class="co">#&gt; [103] BiocManager_1.30.26     cli_3.6.5               xtable_1.8-4           </span></span>
<span><span class="co">#&gt; [106] systemfonts_1.2.3       jquerylib_0.1.4         iSEEu_1.20.0           </span></span>
<span><span class="co">#&gt; [109] Rcpp_1.1.0              triebeard_0.4.1         png_0.1-8              </span></span>
<span><span class="co">#&gt; [112] parallel_4.5.1          pkgdown_2.1.3           ggplot2_3.5.2          </span></span>
<span><span class="co">#&gt; [115] blob_1.2.4              viridisLite_0.4.2       scales_1.4.0           </span></span>
<span><span class="co">#&gt; [118] purrr_1.1.0             crayon_1.5.3            GetoptLong_1.0.5       </span></span>
<span><span class="co">#&gt; [121] rlang_1.1.6             KEGGREST_1.48.1         shinyjs_2.1.0</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Alasoo2018" class="csl-entry">
Alasoo, Kaur, Julia Rodrigues, Subhankar Mukhopadhyay, Andrew J.
Knights, Alice L. Mann, Kousik Kundu, Christine Hale, Gordon Dougan, and
Daniel J. Gaffney. 2018. <span>“<span class="nocase">Shared genetic
effects on chromatin and gene expression indicate a role for enhancer
priming in immune response</span>.”</span> <em>Nature Genetics</em> 50
(3): 424–31. <a href="https://doi.org/10.1038/s41588-018-0046-7" class="external-link">https://doi.org/10.1038/s41588-018-0046-7</a>.
</div>
<div id="ref-Love2014" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“<span class="nocase">Moderated estimation of fold change and dispersion for
RNA-seq data with DESeq2</span>.”</span> <em>Genome Biology</em> 15
(12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8" class="external-link">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-Rigby2023" class="csl-entry">
Rigby, Rachel, Kevin Rue-Albrecht, David Sims, and Jan Rehwinkel. 2023.
<span>“<span class="nocase">Single-cell analysis of signalling responses
to type I interferons</span>.”</span> <em>bioRxiv</em>, 1–68.
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:kevinrue67@gmail.com" class="email">kevinrue67@gmail.com</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="mailto:charlottesoneson@gmail.com" class="email">charlottesoneson@gmail.com</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p><a href="mailto:infinite.monkeys.with.keyboards@gmail.com" class="email">infinite.monkeys.with.keyboards@gmail.com</a><a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p><a href="mailto:abassina@uni-mainz.de" class="email">abassina@uni-mainz.de</a><a href="#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Federico Marini, Kevin Rue-Albrecht, Charlotte Soneson, Aaron Lun, Najla Abassi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
